src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaJsonWriter.cs

```diff
@@ -0,0 +1,1292 @@
+            this.jsonWriter.WriteOptionalProperty("$MaxLength", reference.MaxLength);
+
+            // The value of $Unicode is one of the Boolean literals true or false.Absence of the member means true.
+            this.jsonWriter.WriteOptionalProperty("$Unicode", reference.IsUnicode, CsdlConstants.Default_IsUnicode);
+        }
+
+        internal override void WriteTemporalTypeAttributes(IEdmTemporalTypeReference reference)
+        {
+            // The value of $Precision is a number. Absence of $Precision means arbitrary precision.
+            this.jsonWriter.WriteOptionalProperty("$Precision", reference.Precision);
+        }
+
+        internal override void WriteAnnotationStringAttribute(IEdmDirectValueAnnotation annotation)
+        {
+            var edmValue = (IEdmPrimitiveValue)annotation.Value;
+            if (edmValue != null)
+            {
+                this.jsonWriter.WriteRequiredProperty(annotation.Name, EdmValueWriter.PrimitiveValueAsXml(edmValue));
+            }
+        }
+
+        internal override void WriteAnnotationStringElement(IEdmDirectValueAnnotation annotation)
+        {
+            var edmValue = (IEdmPrimitiveValue)annotation.Value;
+            if (edmValue != null)
+            {
+                this.jsonWriter.WriteStringValue(((IEdmStringValue)edmValue).Value);
+            }
+        }
+
+        internal override void WriteSchemaOperationsHeader<T>(KeyValuePair<string, IList<T>> operation)
+        {
+            // An operation is represented as a member of the schema object whose name is the unqualified name of the operation.
+            this.jsonWriter.WritePropertyName(operation.Key);
+
+            // Whose value is an array
+            this.jsonWriter.WriteStartArray();
+        }
+
+        internal override void WriteSchemaOperationsEnd<T>(KeyValuePair<string, IList<T>> operation)
+        {
+            this.jsonWriter.WriteEndArray();
+        }
+
+        internal override void WriteActionElementHeader(IEdmAction action)
+        {
+            this.jsonWriter.WriteStartObject();
+
+            // The action overload object MUST contain the member $Kind with a string value of Action.
+            this.jsonWriter.WriteRequiredProperty("$Kind", CsdlConstants.Element_Action);
+
+            this.WriteOperationElementAttributes(action);
+        }
+
+        internal override void WriteFunctionElementHeader(IEdmFunction function)
+        {
+            this.jsonWriter.WriteStartObject();
+
+            // The action overload object MUST contain the member $Kind with a string value of Action.
+            this.jsonWriter.WriteRequiredProperty("$Kind", CsdlConstants.Element_Function);
+
+            this.WriteOperationElementAttributes(function);
+
+            if (function.IsComposable)
+            {
+                this.jsonWriter.WriteRequiredProperty("$IsComposable", function.IsComposable, EdmValueWriter.BooleanAsXml);

```

Comment:
>EdmValueWriter.BooleanAsXml [](start = 94, length = 27)

shouldn't this just write true (not BooleanAsXml)?

Reply:
Changed.

---
In reply to: [443764679](https://github.com/OData/odata.net/pull/1775#discussion_r443764679) [](ancestors = 443764679)